sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -0,0 +1,37 @@
+//===---------------- pi2ur.cpp - PI API to UR API  --------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===------------------------------------------------------------------===//
+
+#include <unordered_map>
+
+#include "zer_api.h"
+#include <sycl/detail/pi.h>
+
+// Map of UR error codes to PI error codes
+static pi_result ur2piResult(zer_result_t urResult) {
+  static std::unordered_map<zer_result_t, pi_result> ErrorMapping = {

```

Comment:
Not directly related to the PR:

While we are here, it would be great to replace "global lifetime" objects with a non-trivial d'tor with either pointers to such objects (which would be allocated and dealocated during init and teardown) or objects with trivial d'tor. E.g. for this case we could have an `std::array` with sorted values.

Reply:
I have added this as a TODO note to address it later.